lib/index.js

```diff
@@ -880,10 +880,13 @@ class Image extends RustClass {
       return
     } else if (/^\s*data:/.test(src)) {
       // data URI
-      let split = src.indexOf(','),
-          enc = src.lastIndexOf('base64', split) !== -1 ? 'base64' : 'utf8',
-          content = src.slice(split + 1);
-      data = Buffer.from(content, enc);
+      let [header, mime, enc] = src.match(/^\s*data:(?<mime>[^;]*);(?:charset=)?(?<enc>[^,]*),/) || []
```

Comment:
I ❤️ regex but it's slooooow compared to array-like operations on strings like split or indexOf.  Probably won't break anything here, but just saying...  One optimization is to store the regex as a const somewhere static so it gets optimized after a few uses.  But it's still slow.

Reply:
Love the optimization energy! I know what you mean about regexes being surprisingly slow in tight loops but here I think we can spare a half an ms (given the time that's about to be spent parsing and/or decompressing the image) ;)

FWIW I added a `.slice` before the match to give it a smaller string to work with.